src/relations/db.py

```diff
@@ -319,14 +319,14 @@ def update_endpoints(self, relation: Relation = None) -> None:
         if len(relations) == 0:
             return
 
+        postgresql_version = None
         try:
             postgresql_version = self.charm.postgresql.get_postgresql_version()
```

Comment:
Alternatively we can use `cluster.get_postgresql_version` instead and get the version from the snap.